interpreter/interpreter.go

```diff
@@ -151,6 +156,185 @@ func CompileRegexConstants(regexOptimizations ...*RegexOptimization) Interpretab
 	return decRegexOptimizer(regexOptimizations...)
 }
 
+type typeVerifier func(int64, ...*types.TypeValue) (bool, error)
+
+// CheckStringFormat checks the syntax and cardinality of any string.format calls present in the expression and reports
+// any errors at compile time.
+func CheckStringFormat(enableCheckArgTypes bool, verifier typeVerifier) InterpretableDecorator {
+	return func(inter Interpretable) (Interpretable, error) {
+		call, ok := inter.(InterpretableCall)
+		if !ok {
+			return inter, nil
+		}
+		if call.OverloadID() != "string_format" {
+			return inter, nil
+		}
+		args := call.Args()
+		if len(args) != 2 {
+			return nil, fmt.Errorf("wrong number of arguments to string.format (expected 2, got %d)", len(args))
+		}
+		fmtStrInter, ok := args[0].(InterpretableConst)
+		if !ok {
+			return inter, nil
+		}
+		fmtArgsInter, ok := args[1].(InterpretableConstructor)
+		if !ok {
+			return inter, nil
+		}
+		if fmtArgsInter.Type() != types.ListType {
+			return nil, fmt.Errorf("expected list as second argument to string.format (got %s", fmtArgsInter.Type().TypeName())
```

Comment:
`return iter, nil` here to tolerate `Dyn` values which may, at runtime, turn out to be valid lists?

Reply:
Fixed